### PR TITLE
feat(cli): Add vault ssh certificate credential library

### PR DESCRIPTION
**Pull request by tmessi**

### CODE REVIEW

File: internal/tests/cli/Makefile [hunk 1]
```diff
@@ -1,5 +1,36 @@
...
+TEST_VAULT_PORT ?= 8200
+TEST_VAULT_CONTAINER_NAME ?= boundary-cli-tests-vault
+
+VAULT_ADDR ?= "http://127.0.0.1:$(TEST_VAULT_PORT)"
+VAULT_TOKEN ?= boundarytok
+
+.PHONY: test-vault-up
+test-vault-up:
+	@echo "Using image: $(TEST_VAULT_IMAGE)"
+	@echo " vault addr: $(VAULT_ADDR)"
```
Comment (moduli) [inline internal/tests/cli/Makefile:15]: Extra space?
```suggestion
	@echo "vault addr: $(VAULT_ADDR)"
```

File: internal/tests/cli/boundary/_vault.bash [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    -policy="boundary-controller" \
+    -orphan=true \
+    -period=2m \
+    -renewable=true | \
+    jq -r '.auth.client_token'
+}
+
+function skip_if_no_vault() {
+  if [[ -z $VAULT_ADDR ]] || [[ -z $VAULT_TOKEN ]]; then
+    skip "vault environment variables  \$VAULT_ADDR and \$VAULT_TOKEN not set"
```
Comment (moduli) [inline internal/tests/cli/boundary/_vault.bash:45]: Extra space?
```suggestion
    skip "vault environment variables \$VAULT_ADDR and \$VAULT_TOKEN not set"
```

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -479,6 +486,13 @@ jobs:
...
+          echo "vault version:"
+          vault version
       - name: Run cli bats tests
         run: |
+          export VAULT_ADDR=http://127.0.0.1:8200
+          export VAULT_TOKEN=boundarytok
+          make -C internal/tests/cli test-vault-up
+          until vault status; do docker container inspect boundary-cli-tests-vault &> /dev/null || exit -1; sleep 1; done
           make test-cli
+          make -C internal/tests/cli test-vault-down
```
Comment (moduli) [inline .github/workflows/build.yml:498]: Does this run if a bats test fails?

File: internal/tests/cli/Makefile (hunk 1 above)
Comment (tmessi) [reply internal/tests/cli/Makefile:15]: This is purposeful so the echos all align on the `:`.

File: .github/workflows/build.yml (hunk 3 above)
Comment (tmessi) [reply .github/workflows/build.yml:501]: Split this out into a separate step that will run even on failure.

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -453,6 +453,13 @@ jobs:
           cd /tmp/bats-cli-ui-deps/pass/password-store-1.7.4
           sudo make install
           pass init ${{ secrets.ENOS_GPG_UID }}
+      - name: Download Vault AMD64 binary for integration testing
+        if: steps.dep-cache.outputs.cache-hit != 'true'
+        run: |
+          wget https://releases.hashicorp.com/vault/1.12.2/vault_1.12.2_linux_amd64.zip -O /tmp/bats-cli-ui-deps/vault.zip
```
[the remark this answers is not in the document]
Comment (tmessi) [reply .github/workflows/build.yml:459]: I like that idea, any thoughts on the best way to do that?

[the remark this answers is not in the document]
Comment (moduli) [reply .github/workflows/build.yml:459]: Hm... I was looking at this: https://docs.github.com/en/actions/learn-github-actions/variables#defining-environment-variables-for-a-single-workflow. Hopefully we can insert that into the `wget` url?

**Review by A-440Hz — APPROVED** [review]

File: .github/workflows/build.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (tmessi) [reply .github/workflows/build.yml:459]: Somehow that also needs to work its way into the key used for the caching. I think I will look into this as a follow up.

File: internal/tests/cli/boundary/credential_libraries.bats [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+  [ "$status" -eq 0 ]
+}
+
+@test "boundary/credential-libraries: can not create already created $NEW_VAULT_LIB vault-ssh-certificate libary" {
+  skip_if_no_vault
+
+  local csid=$(credential_store_id $NEW_STORE $DEFAULT_P_ID)
+  run create_vault_ssh_certificate_library \
+    -name $NEW_VAULT_LIB -credential-store-id $csid \
+    -vault-path ssh/sign/foo \
```
Comment (moduli) [inline internal/tests/cli/boundary/credential_libraries.bats:142]: Was it intentional to not have a leading slash here?

**Review by moduli — APPROVED** [review]

**Review by DanielleMiu — APPROVED** [review]

**Pull request merged**